components/eam/src/chemistry/modal_aero/aero_model.F90

```diff
@@ -1741,241 +1741,357 @@ subroutine modal_aero_bcscavcoef_get( m, ncol, isprx, dgn_awet, scavcoefnum, sca
+   real(r8), intent(in)  :: sigmag      ! geometric standard deviation of size distribution
+   real(r8), intent(in)  :: rhoaero     ! aerosol density [g/cm^3]
+   real(r8), intent(in)  :: temp        ! temperature [K]
+   real(r8), intent(in)  :: press       ! pressure [dyne/cm^2]
+   real(r8), intent(out) :: scavratenum, scavratevol  ! scavenging rate for aerosol number and volume [1/hour]
+   integer,  intent(out) :: lunerr      ! logical unit for error message
 
    !   local variables
-   integer nrainsvmax
-   parameter (nrainsvmax=50)
-   real(r8) rrainsv(nrainsvmax), xnumrainsv(nrainsvmax),&
-        vfallrainsv(nrainsvmax)
-
-   integer naerosvmax
-   parameter (naerosvmax=51)
-   real(r8) aaerosv(naerosvmax), &
-     	ynumaerosv(naerosvmax), yvolaerosv(naerosvmax)
-
-   integer i, ja, jr, na, nr
-   real(r8) a, aerodiffus, aeromass, ag0, airdynvisc, airkinvisc
-   real(r8) anumsum, avolsum, cair, chi
-   real(r8) d, dr, dum, dumfuchs, dx
-   real(r8) ebrown, eimpact, eintercept, etotal, freepath
-   real(r8) precip, precipmmhr, precipsum
-   real(r8) r, rainsweepout, reynolds, rhi, rhoair, rlo, rnumsum
-   real(r8) scavsumnum, scavsumnumbb
-   real(r8) scavsumvol, scavsumvolbb
-   real(r8) schmidt, sqrtreynolds, sstar, stokes, sx              
-   real(r8) taurelax, vfall, vfallstp
-   real(r8) x, xg0, xg3, xhi, xlo, xmuwaterair                     
-
-   
+   integer, parameter :: nrainsvmax=50  ! maximum bin number for rain
+   real(r8) :: rrainsv(nrainsvmax)      ! rain radius for each bin [cm]
+   real(r8) :: xnumrainsv(nrainsvmax)   ! rain number for each bin [#/cm3]
+   real(r8) :: vfallrainsv(nrainsvmax)  ! rain falling velocity for each bin [cm/s]
+
+   integer, parameter :: naerosvmax=51  ! maximum bin number for aerosol
+   real(r8) :: raerosv(naerosvmax)      ! aerosol particle radius in each bin [cm] 
+   real(r8) :: fnumaerosv(naerosvmax)   ! fraction of total number in the bin [fraction]
+   real(r8) :: fvolaerosv(naerosvmax)   ! fraction of total volume in the bin [fraction]
+
+   integer  :: ja, jr                   ! index for aerosol and rain
+   integer  :: na, nr                   ! number of aerosol and rain bins
+   real(r8) :: airkinvisc               ! air kinematic viscosity [cm^2/s]
+   real(r8) :: freepath                 ! molecular freepath [cm]
+   real(r8) :: etotal                   ! efficiency of total scavenging effects [fraction]
+   real(r8) :: precip                   ! precipitation rate, fix as 1 mm/hr in this subroutine [cm/s]
+   real(r8) :: r_rain, r_aer            ! rain droplet and aerosol particle radius [cm]
+   real(r8) :: cair                     ! air molar density [dyne/cm^2/erg*mol = mol/cm^3]
+   real(r8) :: rhoair                   ! air mass density [g/cm^3]
+   real(r8) :: rlo, rhi, dr             ! rain droplet bin information [cm]
+   real(r8) :: rainsweepout             ! rain droplet sweep out area [cm3/cm3/s]
```

Comment:
If I were nit-picky, I would say that rainsweepout is 'rain droplet sweep out volume'.

Reply:
yeah, that's a more correct word. Thank you